include/LightGBM/utils/common.h

```diff
@@ -1137,6 +1007,213 @@ class FunctionTimer {
 
 extern Common::Timer global_timer;
 
+
+/*!
+ * Provides locale-independent alternatives to Common's methods.
+ * Essential to make models robust to locale settings.
+ */
+namespace CommonC {
```

Comment:
Add 4 spaces for all this namespace, or extract it out of common namespace for readability. This File is already too dense for code review

Reply:
I also prefer 4 spaces indentation in C++ but LGBM uses 2 spaces everywhere. As this namespace is new, I guess I can make it inconsistent with the rest of the file but well aligned.

I cannot move this code elsewhere though as it's deeply connected to Common. I'd have to add includes everywhere in the codebase and it would be easier for people to forget that changing Common might also require adapting CommonC.

Added indentation though ;)